xc/device.ml

```diff
@@ -1912,6 +1912,30 @@ let stop ~xs ~qemu_domid domid  =
     stop_vgpu ();
     stop_qemu ()
 
+let is_hvm_linux domid =
+	let _XEN_IOPORT_LINUX_PRODNUM = 3 in (* from Linux include/xen/platform_pci.h *)
+	try
+		let open Qmp in
+		match qmp_write_and_read domid (Command (None, Query_xen_platform_pv_driver_info)) with
+		| Some pv_driver_info -> begin match pv_driver_info with
```

Comment:
This could be matched in one step and I think it would be cleaner:

```
| Some(Success (None, Xen_platform_pv_driver_info { product_num; build_num })) ->
  Some ((product_num = _XEN_IOPORT_LINUX_PRODNUM) && (build_num <= 0xff))
| Some(_) -> None (* unexpected? Maybe add some logging here *)
| None -> None
```